var/spack/repos/builtin/packages/viskores/package.py

```diff
@@ -0,0 +1,213 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+import sys
+
+from spack.build_systems.cmake import CMakeBuilder
+from spack.package import *
+
+
+class Viskores(CMakePackage, CudaPackage, ROCmPackage):
+    """Viskores is a toolkit of scientific visualization algorithms for emerging
+    processor architectures. Viskores supports the fine-grained concurrency for
+    data analysis and visualization algorithms required to drive extreme scale
+    computing by providing abstract models for data and execution that can be
+    applied to a variety of algorithms across many different processor
+    architectures."""
+
+    homepage = "https://m.vtk.org://github.com/Viskores/viskores"
+    maintainers("kmorel", "vicentebolea")
+
+    url = "https://github.com/Viskores/Viskores/archive/refs/tags/v1.0.0.tar.gz"
+    git = "https://github.com/Viskores/Viskores.git"
+    tags = ["e4s"]
+
+    test_requires_compiler = True
+
+    version("master", branch="master")
+    version("release", branch="release")
+    version(
+        "1.0.0",
+        sha256="5bff5bbd747b7662bb4630889960371d06fcc5e5a962d974a898d1883f196eba",
+        preferred=True,
+    )
+
+    variant("shared", default=False, description="build shared libs")
+    variant("doubleprecision", default=True, description="enable double precision")
+    variant("logging", default=False, description="build logging support")
+    variant("mpi", default=False, description="build mpi support")
+    variant("rendering", default=True, description="build rendering support")
+    variant("64bitids", default=False, description="enable 64 bits ids")
+    variant("testlib", default=False, description="build test library")
+    variant("fpic", default=False, description="build fpic support")
+    variant("examples", default=True, description="Install builtin examples")
```

Comment:
This is the first time I've given a hard look at these default values in a long while. A lot of them seem out of date. Here are the updated defaults I would suggest.

```suggestion
    variant("shared", default=True, description="build shared libs")
    variant("doubleprecision", default=True, description="enable double precision")
    variant("logging", default=True, description="build logging support")
    variant("mpi", default=True, description="build mpi support")
    variant("rendering", default=True, description="build rendering support")
    variant("64bitids", default=False, description="enable 64 bits ids")
    variant("testlib", default=False, description="build test library")
    variant("fpic", default=False, description="build fpic support")
    variant("examples", default=False, description="Install builtin examples")
```